File: rest-api/src/main/java/space/shefer/receipt/rest/entity/UserProfile.java
package space.shefer.receipt.rest.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Getter
@Setter
@Table(name = "user_profile")
public class UserProfile extends BaseUuidIdEntity {

  @Column(name = "phone", nullable = false, unique = true)
  private String phone;

  @Column(name = "password", nullable = false)
  private String password;

  @Column(name = "fns_request_count")
  private int fnsRequestCount = 0;

  @Column(name = "access_token")
  private String accessToken;

}

File: telegram-bot/src/main/kotlin/space/shefer/receipt/bot/model/UserProfile.kt
package space.shefer.receipt.bot.model

import javax.persistence.*

@Entity
@Table(name = "users")
class UserProfile {

    @Id
    @GeneratedValue
    @Column(name = "id", unique = true, nullable = false)
    var id: Long? = null

    @Column(name = "phone_number", unique = true, nullable = false)
    lateinit var phoneNumber: String

    @Column(name = "first_name", nullable = false)
    lateinit var firstName: String

    @Column(name = "last_name", nullable = false)
    lateinit var lastName: String

}
